crates/nu-command/src/platform/input/input_.rs

```diff
@@ -75,139 +80,97 @@ impl Command for Input {
             ));
         }
 
-        if let Some(bytes_until) = bytes_until {
-            let _ = crossterm::terminal::enable_raw_mode();
-
-            if let Some(prompt) = prompt {
-                print!("{prompt}");
-                let _ = std::io::stdout().flush();
-            }
+        let byte_until = if let Some(bytes_until) = bytes_until {
             if let Some(c) = bytes_until.bytes().next() {
-                let mut buf = [0u8; 1];
-                let mut buffer = vec![];
-
-                let mut stdin = std::io::stdin();
-
-                loop {
-                    if let Err(err) = stdin.read_exact(&mut buf) {
-                        let _ = crossterm::terminal::disable_raw_mode();
-                        return Err(ShellError::IOError(err.to_string()));
-                    }
-                    buffer.push(buf[0]);
-
-                    if i64::try_from(buffer.len()).unwrap_or(0) >= numchar.item {
-                        let _ = crossterm::terminal::disable_raw_mode();
-                        break;
-                    }
-
-                    // 03 symbolizes SIGINT/Ctrl+C
-                    if buf.contains(&3) {
-                        let _ = crossterm::terminal::disable_raw_mode();
-                        return Err(ShellError::IOError("SIGINT".to_string()));
-                    }
-
-                    if buf[0] == c {
-                        let _ = crossterm::terminal::disable_raw_mode();
-                        break;
-                    }
-                }
-
-                Ok(Value::Binary {
-                    val: buffer,
-                    span: call.head,
-                }
-                .into_pipeline_data())
+                Some(c)
             } else {
                 let _ = crossterm::terminal::disable_raw_mode();
-                Err(ShellError::IOError(
+                return Err(ShellError::IOError(
                     "input can't stop on this byte".to_string(),
-                ))
+                ));
```

Comment:
This still reads a bit weird. 

Also not sure about the thing carried over from the original implementation to just take the first byte from whatever unicode codepoint that starts the string passed to the flag. Possibly worth revisiting in a separate PR.

Reply:
For now, it's based on the original implementation and it checks only first byte of until_bytes string.
But I think that "input" command should stop at any byte of until_bytes string.

May I open another issue for discussing this?